app-ios/Sources/CommonComponents/Timetable/TimetableGridCard.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  File.swift
+//  
+//
+//  Created by CHARLES BOND on 2024/08/12.
+//
+
+import Foundation
+
+import SwiftUI
+import Theme
+import class shared.TimetableItem
+
+public struct TimetableGridCard: View {
+    let timetableItem: TimetableItem?
+    let onTap: (TimetableItem) -> Void
+    
+    public init(
+        timetableItem: TimetableItem?,
+        onTap: @escaping (TimetableItem) -> Void
+    ) {
+        self.timetableItem = timetableItem
+        self.onTap = onTap
+    }
+
+    public var body: some View {
+        if let ttItem = timetableItem {
```

Comment:
maybe this line can be replaced with:

```suggestion
        if let timetableItem {
```